src/Storages/MergeTree/MergeTreeSettings.h

```diff
@@ -48,6 +48,7 @@ struct Settings;
     M(UInt64, max_replicated_merges_with_ttl_in_queue, 1, "How many tasks of merging parts with TTL are allowed simultaneously in ReplicatedMergeTree queue.", 0) \
     M(UInt64, number_of_free_entries_in_pool_to_lower_max_size_of_merge, 8, "When there is less than specified number of free entries in pool (or replicated queue), start to lower maximum size of merge to process (or to put in queue). This is to allow small merges to process - not filling the pool with long running merges.", 0) \
     M(UInt64, number_of_free_entries_in_pool_to_execute_mutation, 20, "When there is less than specified number of free entries in pool, do not execute part mutations. This is to leave free threads for regular merges and avoid \"Too many parts\"", 0) \
+    M(UInt64, max_number_of_mutatuins_for_replica, 0, "Limit the number of part mutations per replica to the specified amount. Zero means unlimited (but the execution is still constrained by `background_merges_mutations_concurrency_ratio * background_pool_size` and `number_of_free_entries_in_pool_to_execute_mutation`).", 0) \
```

Comment:
```suggestion
    M(UInt64, max_number_of_mutatuins_for_replica, 0, "Limit the number of part mutations per replica to the specified amount. Zero means no limit on the number of mutations per replica (the execution can still be constrained by other settings).", 0) \
```

is better IMO because being too specific makes it harder to keep it up to date.